File: springboot_export_csv_apache_commons/src/main/java/com/example/demo/helper/CSVHelper.java
package com.example.demo.helper;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Arrays;
import java.util.List;

import org.apache.commons.csv.CSVFormat;
import org.apache.commons.csv.CSVPrinter;
import org.apache.tomcat.util.http.fileupload.ByteArrayOutputStream;

import com.example.demo.model.Employee;

public class CSVHelper {

	public static ByteArrayInputStream employeesToCSV(List<Employee> employees) {
		final CSVFormat format = CSVFormat.DEFAULT.withHeader("ID", "Name", "Email", "Age", "Country", "Role");

		try (

				ByteArrayOutputStream out = new ByteArrayOutputStream();
				CSVPrinter csvPrinter = new CSVPrinter(new PrintWriter(out), format);) {
			for (Employee emp : employees) {
				List<String> data = Arrays.asList(String.valueOf(emp.getId()), emp.getName(), emp.getEmail(),
						String.valueOf(emp.getAge()), emp.getCountry(), emp.getRole());

				csvPrinter.printRecord(data);
			}

			csvPrinter.flush();
			return new ByteArrayInputStream(out.toByteArray());
		} catch (IOException e) {
			throw new RuntimeException("fail to import data to CSV file: " + e.getMessage());
		}
	}
}

File: kotlin-spring-main/kotlin_springboot_apachecommans_exportcsv/src/main/kotlin/com/knf/dev/helper/CSVHelper.kt
package com.knf.dev.helper

import java.io.IOException

import java.io.ByteArrayInputStream

import com.knf.dev.entity.Employee

import java.io.PrintWriter

import org.apache.commons.csv.CSVPrinter

import org.apache.commons.csv.CSVFormat
import java.io.ByteArrayOutputStream
import java.util.*


object CSVHelper {
    fun employeesToCSV(employees: List<Employee>): ByteArrayInputStream {
        val format = CSVFormat.DEFAULT.withHeader("ID", "Name", "Email", "Age", "Country", "Role")
        try {
            ByteArrayOutputStream().use({ out ->
                CSVPrinter(
                    PrintWriter(out),
                    format
                ).use { csvPrinter ->
                    for (emp in employees) {
                        val data: List<String?> = Arrays.asList(
                            emp.id.toString(),
                            emp.name, emp.email,
                            java.lang.String.valueOf(emp.age), emp.country,
                            emp.role
                        )
                        csvPrinter.printRecord(data)
                    }
                    csvPrinter.flush()
                    return ByteArrayInputStream(out.toByteArray())
                }
            })
        } catch (e: IOException) {
            throw RuntimeException(
                "fail to import data to CSV file: "
                        + e.message
            )
        }
    }
}
